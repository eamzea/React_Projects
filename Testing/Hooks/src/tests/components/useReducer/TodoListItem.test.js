import "@testing-library/jest-dom";
import React from "react";
import TodoListItem from "../../../components/useReducer/TodoListItem";
import { shallow } from "enzyme";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Pruebas en <TodoListItem/>", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      ind={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de llamar la función handleDelete", () => {
    wrapper.find("Button").simulate("click");

    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test("Debe de llamar la función handleToggle", () => {
    wrapper.find("p").simulate("click");

    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test("Debe de mostrar el texto correctamente", () => {
    expect(wrapper.find("p").text().trim()).toEqual(
      `${demoTodos[0].id} . ${demoTodos[0].desc}`
    );
  });

  test("Debe de tener la clase complete si está en true", () => {
    const todo = demoTodos[0];
    todo.done = true;

    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        ind={todo.id}
        handleDelete={handleDelete}
        handleToggle={handleToggle}
      />
    );

    expect(wrapper.find("p").hasClass("complete")).toBe(true);
  });
});
