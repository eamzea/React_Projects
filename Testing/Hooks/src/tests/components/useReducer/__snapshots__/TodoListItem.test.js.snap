// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem/> Debe de mostrarse correctamente 1`] = `
<li
  className="list-group-item d-flex align-items-center justify-content-between"
>
  <p
    className="m-0 todos"
    onClick={[Function]}
  >
    1
     . 
    Aprender React
  </p>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    variant="danger"
  >
    Borrar
  </Button>
</li>
`;
